### ISSUE TITLE
support golang's `+incompatible` notation and pseudo-versions

### ISSUE THREAD

**Issue by ovesh** [shoenig/semantic#1]
Currently `Parse()` chokes on a semver like `v2.0.0+incompatible`.
Further, `String()` appends a dash, so it doesn't support the notation either.

**Comment by ovesh** [comment]
`Parse()` also chokes on pseudo-versions like `v0.8.2-0.20190227000051-27936f6d90f9`.

**Comment by shoenig** [comment]
Thanks!

Published 
```
$ taggit minor
taggit: creating tag: v0.1.0
taggit: created tag: v0.1.0
taggit: published tag: v0.1.0
```
